spec/lib/workos/mfa_spec.rb

```diff
@@ -0,0 +1,232 @@
+            authentication_factor_id: 'auth_factor_01FZ4TS0MWPZR7GATS7KCXANQZ',
+          )
+          expect(challenge_factor.authentication_factor_id.class.instance_of?(String))
+        end
+      end
+    end
+    context 'challenge with sms' do
+      it 'returns a challenge factor object for sms' do
+        VCR.use_cassette 'mfa/challenge_factor_sms_valid' do
+          challenge_factor = described_class.challenge_factor(
+            authentication_factor_id: 'auth_factor_01FZ4TS14D1PHFNZ9GF6YD8M1F',
+            sms_template: 'Your code is {{code}}',
+          )
+          expect(challenge_factor.authentication_factor_id.instance_of?(String))
+        end
+      end
+    end
+    context 'challenge with generic' do
+      it 'returns a valid challenge factor object for generic otp' do
+        VCR.use_cassette 'mfa/challenge_factor_generic_valid' do
+          challenge_factor = described_class.challenge_factor(
+            authentication_factor_id: 'auth_factor_01FZ4WMXXA09XF6NK1XMKNWB3M',
+          )
+          expect(challenge_factor.code.instance_of?(String))
+        end
+      end
+    end
+  end
+  describe 'challenge factor with invalid arguments' do
+    context 'challenge with totp mssing authentication_factor_id' do
+      it 'returns argument error' do
+        expect do
+          described_class.challenge_factor
+        end.to raise_error(
+          ArgumentError,
+          "Incomplete arguments: 'authentication_factor_id' is a required argument",
+        )
+      end
+    end
+  end
+  describe 'challenge factor with valid requests' do
+    context 'verify generic otp' do
+      it 'returns a true boolean if the challenge has not been verifed yet' do
+        VCR.use_cassette 'mfa/verify_factor_generic_valid' do
+          verify_factor = described_class.verify_factor(
+            authentication_challenge_id: 'auth_challenge_01FZ4YVRBMXP5ZM0A7BP4AJ12J',
+            code: '897792',
+          )
+          expect(verify_factor.valid == 'true')
+        end
+      end
+    end
+    context 'verify generic otp' do
+      it 'returns error that the challenge has already been verfied' do
+        VCR.use_cassette 'mfa/verify_factor_generic_invalid' do
+          expect do
+            described_class.verify_factor(
+              authentication_challenge_id: 'auth_challenge_01FZ4YVRBMXP5ZM0A7BP4AJ12J',
+              code: '897792',
+            )
+          end.to raise_error(WorkOS::InvalidRequestError)
+        end
+      end
+      context 'verify generic otp' do
+        it 'returns error that the challenge has expired' do
+          VCR.use_cassette 'mfa/verify_factor_generic_expired' do
+            expect do
+              described_class.verify_factor(
+                authentication_challenge_id: 'auth_challenge_01FZ4YVRBMXP5ZM0A7BP4AJ12J',
+                code: '897792',
+              )
+            end.to raise_error(WorkOS::InvalidRequestError)
+          end
+        end
+      end
```

Comment:
So for this case we should add the `code` field to the base error class here: https://github.com/workos/workos-ruby/blob/master/lib/workos/errors.rb#L7

That means we can also remove the `code` and `message` fields from lib/workos/types/verify_factor_struct.rb. Since an exception is getting thrown those fields aren't in use.

Reply:
done!